gpytorch/kernels/rff_kernel.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import math
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from ..lazy import MatmulLazyTensor, RootLazyTensor
+from .kernel import Kernel
+
+
+class RFFKernel(Kernel):
+    r"""
+    Computes a covariance matrix based on Random Fourier Features with the
+    RBFKernel.
+
+    See Random Features for Large-Scale Kernel Machines by Rahimi and Recht
```

Comment:
2 nits:

- Add a year to the citation
- Can you also include a reference for using sin/cos features instead of shifted cos features? Technically Rahimi and Recht do shifted cos features.